### ISSUE TITLE
Cursor jumping in Notebook on OSX

### ISSUE THREAD

**Issue by randyzwitch** [JuliaLang/IJulia.jl#119]
Not sure when this started, but recently I've been having an issue with IJulia Notebooks when I hit tab to indent code, it moves 4 spaces...then after typing something seems to kick in and move the cursor back two spaces. And unfortunately, it doesn't seem to happen with 100% consistency, so I'm finding it very difficult to pretty print code.

Anyone else have this issue or know what's going on? I'm using OSX Lion, IPython through Anaconda.

**Comment by stevengj** [comment]
Seems resolved.

**Comment by randyzwitch** [comment]
It actually happens to me everyday across multiple computers on OSX, I just neglected to keep talking about it

**Comment by stevengj** [comment]
@randyzwitch, oh you're right, I can reproduce.   I just assumed that this is the way CodeMirror works.

**Comment by JobJob** [comment]
Can anyone provide reproduction steps for this? I recently fixed something possibly related. I use [this vim extension](https://github.com/lambdalisue/jupyter-vim-binding) - and piggy backed it to customise my number of spaces per indent/tab. The indentation changes weren't taking effect in cells created after the notebook was loaded (e.g. after running with alt-enter) due to the way the extension was being activated.

**Comment by randyzwitch** [comment]
There isn't a whole lot to reproduce...write some code, say writing a function across multiple lines, using tabs to align the function arguments. You'll start typing, then the whole line will "un-tab" itself, moving all the way to the left.

It literally happens to me dozens of times per hour, but I still use the Notebook because it's such a great way to work.

**Comment by malmaud** [comment]
Might be an issue with the Julia syntax file, which contains the rules for when to indent and unindent. Is there a syntax pattern that you think reliably results in incorrect unindentation?

**Comment by randyzwitch** [comment]
Sorry for not getting back to you more quickly.

Given this code, which is tabbed in for aligning the function arguments, as soon as I start typing a letter between the brackets in the "streams" array, 

```julia
#Start with this
a.signals = [Dict{Any, Any}("name" => "tooltip", 
                            "init" => Dict{Any, Any}(), 
                            "streams" => []
                            )
            ]

#End up with this
a.signals = [Dict{Any, Any}("name" => "tooltip", 
                            "init" => Dict{Any, Any}(), 
    "streams" => [d]
                            )
            ]

```

It would seem that the code is almost trying to enforce the proper tabbing for Python, as the line gets shifted to be one tab in from the line above it, which is undesirable in Julia since whitespace doesn't matter.

**Comment by cstjean** [comment]
I have the same issue, on OSX too. It's always been that way since I started with Julia last June. `function`, `while`, `for`, `if` are indented properly, but multi-line `=`, function calls, and lists are indented to 0 or to one tab, and Jupyter reindents the line at every new word, making it very unpleasant to work with. It's consistent for me, not random.

I tried deleting `profile_julia` and rebuilding IJulia, no luck.

**Comment by stevengj** [comment]
This is a bug with the Julia pygments mode and needs to be fixed there.

**Comment by cstjean** [comment]
Shouldn't it be in CodeMirror? I thought that [julia.js](https://github.com/codemirror/CodeMirror/blob/be7612de2d879e9442025698d9e93b39d2f0749f/mode/julia/julia.js) was the Julia syntax file that applied to Jupyter notebooks.

**Comment by pabloferz** [comment]
As mentioned here https://github.com/codemirror/CodeMirror/issues/4266#issuecomment-250628500 some of this problems work better or are fixed since version 5.12.0 of CodeMirror, the problem is that the version of CodeMirror used by Jupyter is an older one.

**Comment by stevengj** [comment]
Seems no longer relevant?
